### DISCUSSION TITLE
JSON Schema in swagger response

### DISCUSSION THREAD

**Discussion by rupinder10** [go-swagger/go-swagger#2823] [Q&A]
I have an API that returns a body that corresponds to a JSON Schema, The JSON schema structure is included in the API spec file. But when I try to generate the client I get the following error:

name field ArraySchema
package field models
creating generated file "array_schema.go" in "models" as definition
executed template asset:model
source formatting failed on template-generated source ("models\array_schema.go" for definition). Check that your template produces valid code
unformatted generated source "array_schema.go" has been dumped for template debugging purposes. DO NOT build on this source!
source formatting on generated source "definition" failed: models\array_schema.go:118:34: expected ';', found string (and 10 more errors)

The spec file is attached. Can anybody point me what its complaining about ? The error message is very cryptic,

[ServiceManagementBody.zip](https://github.com/go-swagger/go-swagger/files/9485609/ServiceManagementBody.zip)

**Comment by casualjim** [comment]
The error is about this property:

```go
getDollar RefField string
```

which is about this property in the spec

```json
 "get$ref": {
          "type": "string"
        },
```

**Comment by rupinder10** [comment]
Thanks. Removing this property removes the error. Is there a way to handle properties like this ?

**Reply by casualjim** [reply]
You can try just removing the `$`

**Reply by rupinder10** [reply]
Unfortunately I cannot change the schema. Is there a mechanism while generating the code for it to understand the $ is a part of the name and escape it somehow ?

**Reply by rupinder10** [reply]
@casualjim Can you please let me know if it is possible to generate valid code without having to change this schema ?

**Reply by casualjim** [reply]
not without changing the go-swagger code.

One possible way to go about this is using the mixin command to rewrite the spec prior to generating

**Reply by rupinder10** [reply]
Thanks. I will read up on the mixin command.
